src/clients/twitch.ts

```diff
@@ -12,128 +15,117 @@ function registerTwitch(): void
+        let tokenData: AuthResponse;
+
+        try
+        {
+            if(configFile.DEV_TWITCH_TOKEN)
+                tokenData = JSON.parse(await fs.readFile('./tokens.json', 'utf-8'));
+        }
+        catch(error: unknown)
+        {
+            const res: AuthResponse = await authenticateTwitch({
+                scope: configFile.T2D_SCOPE,
+                redirect_uri: configFile.T2D_REDIRECT_URI,
+                client_id: configFile.T2D_CLIENT_ID,
+                client_secret: configFile.T2D_SECRET,
+                use_https: configFile.T2D_HTTPS.enabled
             });
 
-        bridge.twitchClient = tmijs.Client({
-            options: { debug: true },
-            identity: {
-                username: configFile.T2D_USER,
-                password: 'oauth:' + access_token
-            },
-            channels: configFile.T2D_CHANNELS
-        });
+            await fs.writeFile('./tokens.json', JSON.stringify(res));
 
-        bridge.twitchClient.connect().then(() =>
+            loginToTwitch();
+            return;
+        }
+
+        // If bad data is given to our auth provider you'll get a log along the lines of "no valid token avaiable; trying to refresh.." etc.
+        const authProvider: RefreshingAuthProvider = new RefreshingAuthProvider(
+            {
+                'clientId': configFile.T2D_CLIENT_ID,
+                'clientSecret': configFile.T2D_SECRET,
+                onRefresh: async newTokenData => await fs.writeFile('./tokens.json', JSON.stringify(newTokenData, null, 4), 'utf-8')
+            },
+            tokenData! as any || await authenticateTwitch({
+                scope: configFile.T2D_SCOPE,
+                redirect_uri: configFile.T2D_REDIRECT_URI,
+                client_id: configFile.T2D_CLIENT_ID,
+                client_secret: configFile.T2D_SECRET,
+                use_https: configFile.T2D_HTTPS.enabled
+            })
+        );
+
+        bridge.twitch.authChatClient = new ChatClient({ authProvider, channels: [configFile.T2D_CHANNELS[0]] });
+        bridge.twitch.anonChatClient = new ChatClient({ authProvider: undefined, channels: [configFile.T2D_CHANNELS[0]] });
+
+        bridge.twitch.authChatClient.connect().then(() =>
         {
-            console.log('Twitch bot is live! Sending a buffer message...', configFile.T2D_USER);
+            console.log('AUTHED: Twitch bot is live! Sending a buffer message...', configFile.T2D_USER);
             //Send a buffer message that allows us to track messages being sent as the twitch bot
             //TODO: if we need to scale this to *much* more than just one twitch channel, this won't be usable, there will need to be another approach to record the ID's of the bot user
-            bridge.twitchClient!.say(configFile.T2D_CHANNELS[0], 'twitch bot buffer message!');
+            bridge.twitch.authChatClient!.say(configFile.T2D_CHANNELS[0], 'twitch bot buffer message!');
```

Comment:
This is actually one of the things that was used to tackle the trailing ID bug; if it doesn't exist here, wer're good to remove this!

Reply:
Tried not to remove anything which was not necessary to remove; if I'm recalling correctly that would not even send to begin with. Will remove. 

Also "we're"*